# Snapshot report for `test/prefer-await.mjs`

The actual snapshot is saved in `prefer-await.mjs.snap`.

Generated by [AVA](https://avajs.dev).

## Invalid #1
      1 | promise.then(() => {})

> Error 1/1

    `␊
    > 1 | promise.then(() => {})␊
        |         ^^^^ Do not use \`Promise#then(…)\`.␊
    ␊
    --------------------------------------------------------------------------------␊
    Suggestion 1/1: Use \`await\` expression (UNSAFE).␊
      1 | await promise;␊
      2 | {}␊
    `

## Invalid #2
      1 | promise.then(foo)

> Error 1/1

    `␊
    > 1 | promise.then(foo)␊
        |         ^^^^ Do not use \`Promise#then(…)\`.␊
    ␊
    --------------------------------------------------------------------------------␊
    Suggestion 1/1: Use \`await\` expression (UNSAFE).␊
      1 | foo(await promise)␊
    `

## Invalid #3
      1 | promise.then(null, () => {})

> Error 1/1

    `␊
    > 1 | promise.then(null, () => {})␊
        |         ^^^^ Do not use \`Promise#then(…)\`.␊
    `

## Invalid #4
      1 | promise.then(null, foo)

> Error 1/1

    `␊
    > 1 | promise.then(null, foo)␊
        |         ^^^^ Do not use \`Promise#then(…)\`.␊
    `

## Invalid #5
      1 | promise.catch(() => {})

> Error 1/1

    `␊
    > 1 | promise.catch(() => {})␊
        |         ^^^^^ Do not use \`Promise#catch(…)\`.␊
    `

## Invalid #6
      1 | promise.catch(foo)

> Error 1/1

    `␊
    > 1 | promise.catch(foo)␊
        |         ^^^^^ Do not use \`Promise#catch(…)\`.␊
    `

## Invalid #7
      1 | promise.finally(() => {})

> Error 1/1

    `␊
    > 1 | promise.finally(() => {})␊
        |         ^^^^^^^ Do not use \`Promise#finally(…)\`.␊
    `

## Invalid #8
      1 | promise.finally(foo)

> Error 1/1

    `␊
    > 1 | promise.finally(foo)␊
        |         ^^^^^^^ Do not use \`Promise#finally(…)\`.␊
    `

## Invalid #9
      1 | promise.then(foo)

> Error 1/1

    `␊
    > 1 | promise.then(foo)␊
        |         ^^^^ Do not use \`Promise#then(…)\`.␊
    ␊
    --------------------------------------------------------------------------------␊
    Suggestion 1/1: Use \`await\` expression (UNSAFE).␊
      1 | foo(await promise)␊
    `

## Invalid #10
      1 | function nonAsyncFunction() {
      2 | 	promise.then(foo)
      3 | }

> Error 1/1

    `␊
      1 | function nonAsyncFunction() {␊
    > 2 | 	promise.then(foo)␊
        | 	        ^^^^ Do not use \`Promise#then(…)\`.␊
      3 | }␊
    `

## Invalid #11
      1 | async function asyncFunction() {
      2 | 	promise.then(foo)
      3 | }

> Error 1/1

    `␊
      1 | async function asyncFunction() {␊
    > 2 | 	promise.then(foo)␊
        | 	        ^^^^ Do not use \`Promise#then(…)\`.␊
      3 | }␊
    ␊
    --------------------------------------------------------------------------------␊
    Suggestion 1/1: Use \`await\` expression (UNSAFE).␊
      1 | async function asyncFunction() {␊
      2 | 	foo(await promise)␊
      3 | }␊
    `

## Invalid #12
      1 | promise.then(foo);

> Error 1/1

    `␊
    > 1 | promise.then(foo);␊
        |         ^^^^ Do not use \`Promise#then(…)\`.␊
    ␊
    --------------------------------------------------------------------------------␊
    Suggestion 1/1: Use \`await\` expression (UNSAFE).␊
      1 | foo(await promise);␊
    `

## Invalid #13
      1 | function nonAsyncFunction() {
      2 | 	promise.then(foo);
      3 | }

> Error 1/1

    `␊
      1 | function nonAsyncFunction() {␊
    > 2 | 	promise.then(foo);␊
        | 	        ^^^^ Do not use \`Promise#then(…)\`.␊
      3 | }␊
    `

## Invalid #14
      1 | async function asyncFunction() {
      2 | 	promise.then(foo);
      3 | }

> Error 1/1

    `␊
      1 | async function asyncFunction() {␊
    > 2 | 	promise.then(foo);␊
        | 	        ^^^^ Do not use \`Promise#then(…)\`.␊
      3 | }␊
    ␊
    --------------------------------------------------------------------------------␊
    Suggestion 1/1: Use \`await\` expression (UNSAFE).␊
      1 | async function asyncFunction() {␊
      2 | 	foo(await promise);␊
      3 | }␊
    `

## Invalid #15
      1 | promise.then(foo.bar);

> Error 1/1

    `␊
    > 1 | promise.then(foo.bar);␊
        |         ^^^^ Do not use \`Promise#then(…)\`.␊
    ␊
    --------------------------------------------------------------------------------␊
    Suggestion 1/1: Use \`await\` expression (UNSAFE).␊
      1 | foo.bar(await promise);␊
    `

## Invalid #16
      1 | function nonAsyncFunction() {
      2 | 	promise.then(foo.bar);
      3 | }

> Error 1/1

    `␊
      1 | function nonAsyncFunction() {␊
    > 2 | 	promise.then(foo.bar);␊
        | 	        ^^^^ Do not use \`Promise#then(…)\`.␊
      3 | }␊
    `

## Invalid #17
      1 | async function asyncFunction() {
      2 | 	promise.then(foo.bar);
      3 | }

> Error 1/1

    `␊
      1 | async function asyncFunction() {␊
    > 2 | 	promise.then(foo.bar);␊
        | 	        ^^^^ Do not use \`Promise#then(…)\`.␊
      3 | }␊
    ␊
    --------------------------------------------------------------------------------␊
    Suggestion 1/1: Use \`await\` expression (UNSAFE).␊
      1 | async function asyncFunction() {␊
      2 | 	foo.bar(await promise);␊
      3 | }␊
    `

## Invalid #18
      1 | promise.then( (( foo )) )

> Error 1/1

    `␊
    > 1 | promise.then( (( foo )) )␊
        |         ^^^^ Do not use \`Promise#then(…)\`.␊
    ␊
    --------------------------------------------------------------------------------␊
    Suggestion 1/1: Use \`await\` expression (UNSAFE).␊
      1 | (( foo ))(await promise)␊
    `

## Invalid #19
      1 | function nonAsyncFunction() {
      2 | 	promise.then( (( foo )) )
      3 | }

> Error 1/1

    `␊
      1 | function nonAsyncFunction() {␊
    > 2 | 	promise.then( (( foo )) )␊
        | 	        ^^^^ Do not use \`Promise#then(…)\`.␊
      3 | }␊
    `

## Invalid #20
      1 | async function asyncFunction() {
      2 | 	promise.then( (( foo )) )
      3 | }

> Error 1/1

    `␊
      1 | async function asyncFunction() {␊
    > 2 | 	promise.then( (( foo )) )␊
        | 	        ^^^^ Do not use \`Promise#then(…)\`.␊
      3 | }␊
    ␊
    --------------------------------------------------------------------------------␊
    Suggestion 1/1: Use \`await\` expression (UNSAFE).␊
      1 | async function asyncFunction() {␊
      2 | 	(( foo ))(await promise)␊
      3 | }␊
    `

## Invalid #21
      1 | promise.then(new Foo)

> Error 1/1

    `␊
    > 1 | promise.then(new Foo)␊
        |         ^^^^ Do not use \`Promise#then(…)\`.␊
    ␊
    --------------------------------------------------------------------------------␊
    Suggestion 1/1: Use \`await\` expression (UNSAFE).␊
      1 | (new Foo)(await promise)␊
    `

## Invalid #22
      1 | function nonAsyncFunction() {
      2 | 	promise.then(new Foo)
      3 | }

> Error 1/1

    `␊
      1 | function nonAsyncFunction() {␊
    > 2 | 	promise.then(new Foo)␊
        | 	        ^^^^ Do not use \`Promise#then(…)\`.␊
      3 | }␊
    `

## Invalid #23
      1 | async function asyncFunction() {
      2 | 	promise.then(new Foo)
      3 | }

> Error 1/1

    `␊
      1 | async function asyncFunction() {␊
    > 2 | 	promise.then(new Foo)␊
        | 	        ^^^^ Do not use \`Promise#then(…)\`.␊
      3 | }␊
    ␊
    --------------------------------------------------------------------------------␊
    Suggestion 1/1: Use \`await\` expression (UNSAFE).␊
      1 | async function asyncFunction() {␊
      2 | 	(new Foo)(await promise)␊
      3 | }␊
    `

## Invalid #24
      1 | promise.then(foo ?? bar)

> Error 1/1

    `␊
    > 1 | promise.then(foo ?? bar)␊
        |         ^^^^ Do not use \`Promise#then(…)\`.␊
    ␊
    --------------------------------------------------------------------------------␊
    Suggestion 1/1: Use \`await\` expression (UNSAFE).␊
      1 | (foo ?? bar)(await promise)␊
    `

## Invalid #25
      1 | function nonAsyncFunction() {
      2 | 	promise.then(foo ?? bar)
      3 | }

> Error 1/1

    `␊
      1 | function nonAsyncFunction() {␊
    > 2 | 	promise.then(foo ?? bar)␊
        | 	        ^^^^ Do not use \`Promise#then(…)\`.␊
      3 | }␊
    `

## Invalid #26
      1 | async function asyncFunction() {
      2 | 	promise.then(foo ?? bar)
      3 | }

> Error 1/1

    `␊
      1 | async function asyncFunction() {␊
    > 2 | 	promise.then(foo ?? bar)␊
        | 	        ^^^^ Do not use \`Promise#then(…)\`.␊
      3 | }␊
    ␊
    --------------------------------------------------------------------------------␊
    Suggestion 1/1: Use \`await\` expression (UNSAFE).␊
      1 | async function asyncFunction() {␊
      2 | 	(foo ?? bar)(await promise)␊
      3 | }␊
    `

## Invalid #27
      1 | const foo = []
      2 | promise.then([foo][0])

> Error 1/1

    `␊
      1 | const foo = []␊
    > 2 | promise.then([foo][0])␊
        |         ^^^^ Do not use \`Promise#then(…)\`.␊
    ␊
    --------------------------------------------------------------------------------␊
    Suggestion 1/1: Use \`await\` expression (UNSAFE).␊
      1 | const foo = []␊
      2 | ;[foo][0](await promise)␊
    `

## Invalid #28
      1 | function nonAsyncFunction() {
      2 | 	const foo = []
      3 | 	promise.then([foo][0])
      4 | }

> Error 1/1

    `␊
      1 | function nonAsyncFunction() {␊
      2 | 	const foo = []␊
    > 3 | 	promise.then([foo][0])␊
        | 	        ^^^^ Do not use \`Promise#then(…)\`.␊
      4 | }␊
    `

## Invalid #29
      1 | async function asyncFunction() {
      2 | 	const foo = []
      3 | 	promise.then([foo][0])
      4 | }

> Error 1/1

    `␊
      1 | async function asyncFunction() {␊
      2 | 	const foo = []␊
    > 3 | 	promise.then([foo][0])␊
        | 	        ^^^^ Do not use \`Promise#then(…)\`.␊
      4 | }␊
    ␊
    --------------------------------------------------------------------------------␊
    Suggestion 1/1: Use \`await\` expression (UNSAFE).␊
      1 | async function asyncFunction() {␊
      2 | 	const foo = []␊
      3 | 	;[foo][0](await promise)␊
      4 | }␊
    `

## Invalid #30
      1 | promise.then(() => {})

> Error 1/1

    `␊
    > 1 | promise.then(() => {})␊
        |         ^^^^ Do not use \`Promise#then(…)\`.␊
    ␊
    --------------------------------------------------------------------------------␊
    Suggestion 1/1: Use \`await\` expression (UNSAFE).␊
      1 | await promise;␊
      2 | {}␊
    `

## Invalid #31
      1 | function nonAsyncFunction() {
      2 | 	promise.then(() => {})
      3 | }

> Error 1/1

    `␊
      1 | function nonAsyncFunction() {␊
    > 2 | 	promise.then(() => {})␊
        | 	        ^^^^ Do not use \`Promise#then(…)\`.␊
      3 | }␊
    `

## Invalid #32
      1 | async function asyncFunction() {
      2 | 	promise.then(() => {})
      3 | }

> Error 1/1

    `␊
      1 | async function asyncFunction() {␊
    > 2 | 	promise.then(() => {})␊
        | 	        ^^^^ Do not use \`Promise#then(…)\`.␊
      3 | }␊
    ␊
    --------------------------------------------------------------------------------␊
    Suggestion 1/1: Use \`await\` expression (UNSAFE).␊
      1 | async function asyncFunction() {␊
      2 | 	await promise;␊
      3 | {}␊
      4 | }␊
    `

## Invalid #33
      1 | promise.then(() => ({}))

> Error 1/1

    `␊
    > 1 | promise.then(() => ({}))␊
        |         ^^^^ Do not use \`Promise#then(…)\`.␊
    ␊
    --------------------------------------------------------------------------------␊
    Suggestion 1/1: Use \`await\` expression (UNSAFE).␊
      1 | await promise;␊
      2 | ({})␊
    `

## Invalid #34
      1 | function nonAsyncFunction() {
      2 | 	promise.then(() => ({}))
      3 | }

> Error 1/1

    `␊
      1 | function nonAsyncFunction() {␊
    > 2 | 	promise.then(() => ({}))␊
        | 	        ^^^^ Do not use \`Promise#then(…)\`.␊
      3 | }␊
    `

## Invalid #35
      1 | async function asyncFunction() {
      2 | 	promise.then(() => ({}))
      3 | }

> Error 1/1

    `␊
      1 | async function asyncFunction() {␊
    > 2 | 	promise.then(() => ({}))␊
        | 	        ^^^^ Do not use \`Promise#then(…)\`.␊
      3 | }␊
    ␊
    --------------------------------------------------------------------------------␊
    Suggestion 1/1: Use \`await\` expression (UNSAFE).␊
      1 | async function asyncFunction() {␊
      2 | 	await promise;␊
      3 | ({})␊
      4 | }␊
    `

## Invalid #36
      1 | promise.then(() => {
      2 | 	return []
      3 | })

> Error 1/1

    `␊
    > 1 | promise.then(() => {␊
        |         ^^^^ Do not use \`Promise#then(…)\`.␊
      2 | 	return []␊
      3 | })␊
    `

## Invalid #37
      1 | function nonAsyncFunction() {
      2 | 	promise.then(() => {
      3 | 		return []
      4 | 	})
      5 | }

> Error 1/1

    `␊
      1 | function nonAsyncFunction() {␊
    > 2 | 	promise.then(() => {␊
        | 	        ^^^^ Do not use \`Promise#then(…)\`.␊
      3 | 		return []␊
      4 | 	})␊
      5 | }␊
    `

## Invalid #38
      1 | async function asyncFunction() {
      2 | 	promise.then(() => {
      3 | 		return []
      4 | 	})
      5 | }

> Error 1/1

    `␊
      1 | async function asyncFunction() {␊
    > 2 | 	promise.then(() => {␊
        | 	        ^^^^ Do not use \`Promise#then(…)\`.␊
      3 | 		return []␊
      4 | 	})␊
      5 | }␊
    `

## Invalid #39
      1 | promise.then(() => {
      2 | 	function a() {
      3 | 		return []
      4 | 	}
      5 | })

> Error 1/1

    `␊
    > 1 | promise.then(() => {␊
        |         ^^^^ Do not use \`Promise#then(…)\`.␊
      2 | 	function a() {␊
      3 | 		return []␊
      4 | 	}␊
      5 | })␊
    ␊
    --------------------------------------------------------------------------------␊
    Suggestion 1/1: Use \`await\` expression (UNSAFE).␊
      1 | await promise;␊
      2 | {␊
      3 | 	function a() {␊
      4 | 		return []␊
      5 | 	}␊
      6 | }␊
    `

## Invalid #40
      1 | function nonAsyncFunction() {
      2 | 	promise.then(() => {
      3 | 		function a() {
      4 | 			return []
      5 | 		}
      6 | 	})
      7 | }

> Error 1/1

    `␊
      1 | function nonAsyncFunction() {␊
    > 2 | 	promise.then(() => {␊
        | 	        ^^^^ Do not use \`Promise#then(…)\`.␊
      3 | 		function a() {␊
      4 | 			return []␊
      5 | 		}␊
      6 | 	})␊
      7 | }␊
    `

## Invalid #41
      1 | async function asyncFunction() {
      2 | 	promise.then(() => {
      3 | 		function a() {
      4 | 			return []
      5 | 		}
      6 | 	})
      7 | }

> Error 1/1

    `␊
      1 | async function asyncFunction() {␊
    > 2 | 	promise.then(() => {␊
        | 	        ^^^^ Do not use \`Promise#then(…)\`.␊
      3 | 		function a() {␊
      4 | 			return []␊
      5 | 		}␊
      6 | 	})␊
      7 | }␊
    ␊
    --------------------------------------------------------------------------------␊
    Suggestion 1/1: Use \`await\` expression (UNSAFE).␊
      1 | async function asyncFunction() {␊
      2 | 	await promise;␊
      3 | {␊
      4 | 		function a() {␊
      5 | 			return []␊
      6 | 		}␊
      7 | 	}␊
      8 | }␊
    `

## Invalid #42
      1 | promise.then((a) => {})

> Error 1/1

    `␊
    > 1 | promise.then((a) => {})␊
        |         ^^^^ Do not use \`Promise#then(…)\`.␊
    ␊
    --------------------------------------------------------------------------------␊
    Suggestion 1/1: Use \`await\` expression (UNSAFE).␊
      1 | let a = await promise;␊
      2 | {}␊
    `

## Invalid #43
      1 | function nonAsyncFunction() {
      2 | 	promise.then((a) => {})
      3 | }

> Error 1/1

    `␊
      1 | function nonAsyncFunction() {␊
    > 2 | 	promise.then((a) => {})␊
        | 	        ^^^^ Do not use \`Promise#then(…)\`.␊
      3 | }␊
    `

## Invalid #44
      1 | async function asyncFunction() {
      2 | 	promise.then((a) => {})
      3 | }

> Error 1/1

    `␊
      1 | async function asyncFunction() {␊
    > 2 | 	promise.then((a) => {})␊
        | 	        ^^^^ Do not use \`Promise#then(…)\`.␊
      3 | }␊
    ␊
    --------------------------------------------------------------------------------␊
    Suggestion 1/1: Use \`await\` expression (UNSAFE).␊
      1 | async function asyncFunction() {␊
      2 | 	let a = await promise;␊
      3 | {}␊
      4 | }␊
    `

## Invalid #45
      1 | promise.then(({a}) => {})

> Error 1/1

    `␊
    > 1 | promise.then(({a}) => {})␊
        |         ^^^^ Do not use \`Promise#then(…)\`.␊
    ␊
    --------------------------------------------------------------------------------␊
    Suggestion 1/1: Use \`await\` expression (UNSAFE).␊
      1 | let {a} = await promise;␊
      2 | {}␊
    `

## Invalid #46
      1 | function nonAsyncFunction() {
      2 | 	promise.then(({a}) => {})
      3 | }

> Error 1/1

    `␊
      1 | function nonAsyncFunction() {␊
    > 2 | 	promise.then(({a}) => {})␊
        | 	        ^^^^ Do not use \`Promise#then(…)\`.␊
      3 | }␊
    `

## Invalid #47
      1 | async function asyncFunction() {
      2 | 	promise.then(({a}) => {})
      3 | }

> Error 1/1

    `␊
      1 | async function asyncFunction() {␊
    > 2 | 	promise.then(({a}) => {})␊
        | 	        ^^^^ Do not use \`Promise#then(…)\`.␊
      3 | }␊
    ␊
    --------------------------------------------------------------------------------␊
    Suggestion 1/1: Use \`await\` expression (UNSAFE).␊
      1 | async function asyncFunction() {␊
      2 | 	let {a} = await promise;␊
      3 | {}␊
      4 | }␊
    `

## Invalid #48
      1 | promise.then(([]) => {})

> Error 1/1

    `␊
    > 1 | promise.then(([]) => {})␊
        |         ^^^^ Do not use \`Promise#then(…)\`.␊
    ␊
    --------------------------------------------------------------------------------␊
    Suggestion 1/1: Use \`await\` expression (UNSAFE).␊
      1 | await promise;␊
      2 | {}␊
    `

## Invalid #49
      1 | function nonAsyncFunction() {
      2 | 	promise.then(([]) => {})
      3 | }

> Error 1/1

    `␊
      1 | function nonAsyncFunction() {␊
    > 2 | 	promise.then(([]) => {})␊
        | 	        ^^^^ Do not use \`Promise#then(…)\`.␊
      3 | }␊
    `

## Invalid #50
      1 | async function asyncFunction() {
      2 | 	promise.then(([]) => {})
      3 | }

> Error 1/1

    `␊
      1 | async function asyncFunction() {␊
    > 2 | 	promise.then(([]) => {})␊
        | 	        ^^^^ Do not use \`Promise#then(…)\`.␊
      3 | }␊
    ␊
    --------------------------------------------------------------------------------␊
    Suggestion 1/1: Use \`await\` expression (UNSAFE).␊
      1 | async function asyncFunction() {␊
      2 | 	await promise;␊
      3 | {}␊
      4 | }␊
    `

## Invalid #51
      1 | promise.then(({}) => {})

> Error 1/1

    `␊
    > 1 | promise.then(({}) => {})␊
        |         ^^^^ Do not use \`Promise#then(…)\`.␊
    ␊
    --------------------------------------------------------------------------------␊
    Suggestion 1/1: Use \`await\` expression (UNSAFE).␊
      1 | await promise;␊
      2 | {}␊
    `

## Invalid #52
      1 | function nonAsyncFunction() {
      2 | 	promise.then(({}) => {})
      3 | }

> Error 1/1

    `␊
      1 | function nonAsyncFunction() {␊
    > 2 | 	promise.then(({}) => {})␊
        | 	        ^^^^ Do not use \`Promise#then(…)\`.␊
      3 | }␊
    `

## Invalid #53
      1 | async function asyncFunction() {
      2 | 	promise.then(({}) => {})
      3 | }

> Error 1/1

    `␊
      1 | async function asyncFunction() {␊
    > 2 | 	promise.then(({}) => {})␊
        | 	        ^^^^ Do not use \`Promise#then(…)\`.␊
      3 | }␊
    ␊
    --------------------------------------------------------------------------------␊
    Suggestion 1/1: Use \`await\` expression (UNSAFE).␊
      1 | async function asyncFunction() {␊
      2 | 	await promise;␊
      3 | {}␊
      4 | }␊
    `

## Invalid #54
      1 | promise.then(({a}) => {
      2 | 	const b = 1;
      3 | })

> Error 1/1

    `␊
    > 1 | promise.then(({a}) => {␊
        |         ^^^^ Do not use \`Promise#then(…)\`.␊
      2 | 	const b = 1;␊
      3 | })␊
    ␊
    --------------------------------------------------------------------------------␊
    Suggestion 1/1: Use \`await\` expression (UNSAFE).␊
      1 | let {a} = await promise;␊
      2 | {␊
      3 | 	const b = 1;␊
      4 | }␊
    `

## Invalid #55
      1 | function nonAsyncFunction() {
      2 | 	promise.then(({a}) => {
      3 | 		const b = 1;
      4 | 	})
      5 | }

> Error 1/1

    `␊
      1 | function nonAsyncFunction() {␊
    > 2 | 	promise.then(({a}) => {␊
        | 	        ^^^^ Do not use \`Promise#then(…)\`.␊
      3 | 		const b = 1;␊
      4 | 	})␊
      5 | }␊
    `

## Invalid #56
      1 | async function asyncFunction() {
      2 | 	promise.then(({a}) => {
      3 | 		const b = 1;
      4 | 	})
      5 | }

> Error 1/1

    `␊
      1 | async function asyncFunction() {␊
    > 2 | 	promise.then(({a}) => {␊
        | 	        ^^^^ Do not use \`Promise#then(…)\`.␊
      3 | 		const b = 1;␊
      4 | 	})␊
      5 | }␊
    ␊
    --------------------------------------------------------------------------------␊
    Suggestion 1/1: Use \`await\` expression (UNSAFE).␊
      1 | async function asyncFunction() {␊
      2 | 	let {a} = await promise;␊
      3 | {␊
      4 | 		const b = 1;␊
      5 | 	}␊
      6 | }␊
    `

## Invalid #57
      1 | const a = 1, b = 2;
      2 | promise.then(({a}) => {
      3 | 	const b = 1;
      4 | })

> Error 1/1

    `␊
      1 | const a = 1, b = 2;␊
    > 2 | promise.then(({a}) => {␊
        |         ^^^^ Do not use \`Promise#then(…)\`.␊
      3 | 	const b = 1;␊
      4 | })␊
    ␊
    --------------------------------------------------------------------------------␊
    Suggestion 1/1: Use \`await\` expression (UNSAFE).␊
      1 | const a = 1, b = 2;␊
      2 | {␊
      3 | let {a} = await promise;␊
      4 | {␊
      5 | 	const b = 1;␊
      6 | }␊
      7 | }␊
    `

## Invalid #58
      1 | function nonAsyncFunction() {
      2 | 	const a = 1, b = 2;
      3 | 	promise.then(({a}) => {
      4 | 		const b = 1;
      5 | 	})
      6 | }

> Error 1/1

    `␊
      1 | function nonAsyncFunction() {␊
      2 | 	const a = 1, b = 2;␊
    > 3 | 	promise.then(({a}) => {␊
        | 	        ^^^^ Do not use \`Promise#then(…)\`.␊
      4 | 		const b = 1;␊
      5 | 	})␊
      6 | }␊
    `

## Invalid #59
      1 | async function asyncFunction() {
      2 | 	const a = 1, b = 2;
      3 | 	promise.then(({a}) => {
      4 | 		const b = 1;
      5 | 	})
      6 | }

> Error 1/1

    `␊
      1 | async function asyncFunction() {␊
      2 | 	const a = 1, b = 2;␊
    > 3 | 	promise.then(({a}) => {␊
        | 	        ^^^^ Do not use \`Promise#then(…)\`.␊
      4 | 		const b = 1;␊
      5 | 	})␊
      6 | }␊
    ␊
    --------------------------------------------------------------------------------␊
    Suggestion 1/1: Use \`await\` expression (UNSAFE).␊
      1 | async function asyncFunction() {␊
      2 | 	const a = 1, b = 2;␊
      3 | 	{␊
      4 | let {a} = await promise;␊
      5 | {␊
      6 | 		const b = 1;␊
      7 | 	}␊
      8 | }␊
      9 | }␊
    `

## Invalid #60
      1 | (( promise.then(() => {}) ))

> Error 1/1

    `␊
    > 1 | (( promise.then(() => {}) ))␊
        |            ^^^^ Do not use \`Promise#then(…)\`.␊
    ␊
    --------------------------------------------------------------------------------␊
    Suggestion 1/1: Use \`await\` expression (UNSAFE).␊
      1 |  await promise;␊
      2 | {} ␊
    `

## Invalid #61
      1 | function nonAsyncFunction() {
      2 | 	(( promise.then(() => {}) ))
      3 | }

> Error 1/1

    `␊
      1 | function nonAsyncFunction() {␊
    > 2 | 	(( promise.then(() => {}) ))␊
        | 	           ^^^^ Do not use \`Promise#then(…)\`.␊
      3 | }␊
    `

## Invalid #62
      1 | async function asyncFunction() {
      2 | 	(( promise.then(() => {}) ))
      3 | }

> Error 1/1

    `␊
      1 | async function asyncFunction() {␊
    > 2 | 	(( promise.then(() => {}) ))␊
        | 	           ^^^^ Do not use \`Promise#then(…)\`.␊
      3 | }␊
    ␊
    --------------------------------------------------------------------------------␊
    Suggestion 1/1: Use \`await\` expression (UNSAFE).␊
      1 | async function asyncFunction() {␊
      2 | 	 await promise;␊
      3 | {} ␊
      4 | }␊
    `

## Invalid #63
      1 | promise.then(async () => {})

> Error 1/1

    `␊
    > 1 | promise.then(async () => {})␊
        |         ^^^^ Do not use \`Promise#then(…)\`.␊
    `

## Invalid #64
      1 | function nonAsyncFunction() {
      2 | 	promise.then(async () => {})
      3 | }

> Error 1/1

    `␊
      1 | function nonAsyncFunction() {␊
    > 2 | 	promise.then(async () => {})␊
        | 	        ^^^^ Do not use \`Promise#then(…)\`.␊
      3 | }␊
    `

## Invalid #65
      1 | async function asyncFunction() {
      2 | 	promise.then(async () => {})
      3 | }

> Error 1/1

    `␊
      1 | async function asyncFunction() {␊
    > 2 | 	promise.then(async () => {})␊
        | 	        ^^^^ Do not use \`Promise#then(…)\`.␊
      3 | }␊
    `

## Invalid #66
      1 | promise.then(function * () {})

> Error 1/1

    `␊
    > 1 | promise.then(function * () {})␊
        |         ^^^^ Do not use \`Promise#then(…)\`.␊
    `

## Invalid #67
      1 | function nonAsyncFunction() {
      2 | 	promise.then(function * () {})
      3 | }

> Error 1/1

    `␊
      1 | function nonAsyncFunction() {␊
    > 2 | 	promise.then(function * () {})␊
        | 	        ^^^^ Do not use \`Promise#then(…)\`.␊
      3 | }␊
    `

## Invalid #68
      1 | async function asyncFunction() {
      2 | 	promise.then(function * () {})
      3 | }

> Error 1/1

    `␊
      1 | async function asyncFunction() {␊
    > 2 | 	promise.then(function * () {})␊
        | 	        ^^^^ Do not use \`Promise#then(…)\`.␊
      3 | }␊
    `

## Invalid #69
      1 | promise.then(async function * () {})

> Error 1/1

    `␊
    > 1 | promise.then(async function * () {})␊
        |         ^^^^ Do not use \`Promise#then(…)\`.␊
    `

## Invalid #70
      1 | function nonAsyncFunction() {
      2 | 	promise.then(async function * () {})
      3 | }

> Error 1/1

    `␊
      1 | function nonAsyncFunction() {␊
    > 2 | 	promise.then(async function * () {})␊
        | 	        ^^^^ Do not use \`Promise#then(…)\`.␊
      3 | }␊
    `

## Invalid #71
      1 | async function asyncFunction() {
      2 | 	promise.then(async function * () {})
      3 | }

> Error 1/1

    `␊
      1 | async function asyncFunction() {␊
    > 2 | 	promise.then(async function * () {})␊
        | 	        ^^^^ Do not use \`Promise#then(…)\`.␊
      3 | }␊
    `

## Invalid #72
      1 | promise.then((...foo) => {})

> Error 1/1

    `␊
    > 1 | promise.then((...foo) => {})␊
        |         ^^^^ Do not use \`Promise#then(…)\`.␊
    `

## Invalid #73
      1 | function nonAsyncFunction() {
      2 | 	promise.then((...foo) => {})
      3 | }

> Error 1/1

    `␊
      1 | function nonAsyncFunction() {␊
    > 2 | 	promise.then((...foo) => {})␊
        | 	        ^^^^ Do not use \`Promise#then(…)\`.␊
      3 | }␊
    `

## Invalid #74
      1 | async function asyncFunction() {
      2 | 	promise.then((...foo) => {})
      3 | }

> Error 1/1

    `␊
      1 | async function asyncFunction() {␊
    > 2 | 	promise.then((...foo) => {})␊
        | 	        ^^^^ Do not use \`Promise#then(…)\`.␊
      3 | }␊
    `

## Invalid #75
      1 | promise.then((a, extraParameter) => {})

> Error 1/1

    `␊
    > 1 | promise.then((a, extraParameter) => {})␊
        |         ^^^^ Do not use \`Promise#then(…)\`.␊
    `

## Invalid #76
      1 | function nonAsyncFunction() {
      2 | 	promise.then((a, extraParameter) => {})
      3 | }

> Error 1/1

    `␊
      1 | function nonAsyncFunction() {␊
    > 2 | 	promise.then((a, extraParameter) => {})␊
        | 	        ^^^^ Do not use \`Promise#then(…)\`.␊
      3 | }␊
    `

## Invalid #77
      1 | async function asyncFunction() {
      2 | 	promise.then((a, extraParameter) => {})
      3 | }

> Error 1/1

    `␊
      1 | async function asyncFunction() {␊
    > 2 | 	promise.then((a, extraParameter) => {})␊
        | 	        ^^^^ Do not use \`Promise#then(…)\`.␊
      3 | }␊
    `

## Invalid #78
      1 | async function * a() {
      2 | 	promise.then(call);
      3 | }

> Error 1/1

    `␊
      1 | async function * a() {␊
    > 2 | 	promise.then(call);␊
        | 	        ^^^^ Do not use \`Promise#then(…)\`.␊
      3 | }␊
    ␊
    --------------------------------------------------------------------------------␊
    Suggestion 1/1: Use \`await\` expression (UNSAFE).␊
      1 | async function * a() {␊
      2 | 	call(await promise);␊
      3 | }␊
    `

## Invalid #79
      1 | const object = {
      2 | 	a() {
      3 | 		promise.then(call);
      4 | 	}
      5 | }

> Error 1/1

    `␊
      1 | const object = {␊
      2 | 	a() {␊
    > 3 | 		promise.then(call);␊
        | 		        ^^^^ Do not use \`Promise#then(…)\`.␊
      4 | 	}␊
      5 | }␊
    `

## Invalid #80
      1 | const object = {
      2 | 	async a() {
      3 | 		promise.then(call);
      4 | 	}
      5 | }

> Error 1/1

    `␊
      1 | const object = {␊
      2 | 	async a() {␊
    > 3 | 		promise.then(call);␊
        | 		        ^^^^ Do not use \`Promise#then(…)\`.␊
      4 | 	}␊
      5 | }␊
    ␊
    --------------------------------------------------------------------------------␊
    Suggestion 1/1: Use \`await\` expression (UNSAFE).␊
      1 | const object = {␊
      2 | 	async a() {␊
      3 | 		call(await promise);␊
      4 | 	}␊
      5 | }␊
    `

## Invalid #81
      1 | const object = {
      2 | 	async * a() {
      3 | 		promise.then(call);
      4 | 	}
      5 | }

> Error 1/1

    `␊
      1 | const object = {␊
      2 | 	async * a() {␊
    > 3 | 		promise.then(call);␊
        | 		        ^^^^ Do not use \`Promise#then(…)\`.␊
      4 | 	}␊
      5 | }␊
    ␊
    --------------------------------------------------------------------------------␊
    Suggestion 1/1: Use \`await\` expression (UNSAFE).␊
      1 | const object = {␊
      2 | 	async * a() {␊
      3 | 		call(await promise);␊
      4 | 	}␊
      5 | }␊
    `

## Invalid #82
      1 | class A {
      2 | 	a() {
      3 | 		promise.then(call);
      4 | 	}
      5 | }

> Error 1/1

    `␊
      1 | class A {␊
      2 | 	a() {␊
    > 3 | 		promise.then(call);␊
        | 		        ^^^^ Do not use \`Promise#then(…)\`.␊
      4 | 	}␊
      5 | }␊
    `

## Invalid #83
      1 | class A {
      2 | 	async a() {
      3 | 		promise.then(call);
      4 | 	}
      5 | }

> Error 1/1

    `␊
      1 | class A {␊
      2 | 	async a() {␊
    > 3 | 		promise.then(call);␊
        | 		        ^^^^ Do not use \`Promise#then(…)\`.␊
      4 | 	}␊
      5 | }␊
    ␊
    --------------------------------------------------------------------------------␊
    Suggestion 1/1: Use \`await\` expression (UNSAFE).␊
      1 | class A {␊
      2 | 	async a() {␊
      3 | 		call(await promise);␊
      4 | 	}␊
      5 | }␊
    `

## Invalid #84
      1 | class A {
      2 | 	async * a() {
      3 | 		promise.then(call);
      4 | 	}
      5 | }

> Error 1/1

    `␊
      1 | class A {␊
      2 | 	async * a() {␊
    > 3 | 		promise.then(call);␊
        | 		        ^^^^ Do not use \`Promise#then(…)\`.␊
      4 | 	}␊
      5 | }␊
    ␊
    --------------------------------------------------------------------------------␊
    Suggestion 1/1: Use \`await\` expression (UNSAFE).␊
      1 | class A {␊
      2 | 	async * a() {␊
      3 | 		call(await promise);␊
      4 | 	}␊
      5 | }␊
    `

## Invalid #85
      1 | promise.then(call1)
      2 | 	.then(call2)
      3 | 	.then(call3);

> Error 1/3

    `␊
    > 1 | promise.then(call1)␊
        |         ^^^^ Do not use \`Promise#then(…)\`.␊
      2 | 	.then(call2)␊
      3 | 	.then(call3);␊
    `

> Error 2/3

    `␊
      1 | promise.then(call1)␊
    > 2 | 	.then(call2)␊
        | 	 ^^^^ Do not use \`Promise#then(…)\`.␊
      3 | 	.then(call3);␊
    `

> Error 3/3

    `␊
      1 | promise.then(call1)␊
      2 | 	.then(call2)␊
    > 3 | 	.then(call3);␊
        | 	 ^^^^ Do not use \`Promise#then(…)\`.␊
    ␊
    --------------------------------------------------------------------------------␊
    Suggestion 1/1: Use \`await\` expression (UNSAFE).␊
      1 | call3(await promise.then(call1)␊
      2 | 	.then(call2));␊
    `

## Invalid #86
      1 | try {
      2 | 	promise.then(call);
      3 | } catch {}

> Error 1/1

    `␊
      1 | try {␊
    > 2 | 	promise.then(call);␊
        | 	        ^^^^ Do not use \`Promise#then(…)\`.␊
      3 | } catch {}␊
    `

## Invalid #87
      1 | function foo() {
      2 | 	return promise.then(call1)
      3 | }

> Error 1/1

    `␊
      1 | function foo() {␊
    > 2 | 	return promise.then(call1)␊
        | 	               ^^^^ Do not use \`Promise#then(…)\`.␊
      3 | }␊
    `

## Invalid #88
      1 | class A {
      2 | 	a() {
      3 | 		return promise.then(call);
      4 | 	}
      5 | }

> Error 1/1

    `␊
      1 | class A {␊
      2 | 	a() {␊
    > 3 | 		return promise.then(call);␊
        | 		               ^^^^ Do not use \`Promise#then(…)\`.␊
      4 | 	}␊
      5 | }␊
    `

## Invalid #89
      1 | const object = {
      2 | 	a() {
      3 | 		return promise.then(call);
      4 | 	}
      5 | }

> Error 1/1

    `␊
      1 | const object = {␊
      2 | 	a() {␊
    > 3 | 		return promise.then(call);␊
        | 		               ^^^^ Do not use \`Promise#then(…)\`.␊
      4 | 	}␊
      5 | }␊
    `

## Invalid #90
      1 | const object = {
      2 | 	a: function() {
      3 | 		return promise.then(call);
      4 | 	}
      5 | }

> Error 1/1

    `␊
      1 | const object = {␊
      2 | 	a: function() {␊
    > 3 | 		return promise.then(call);␊
        | 		               ^^^^ Do not use \`Promise#then(…)\`.␊
      4 | 	}␊
      5 | }␊
    `
